src/Microsoft.OData.Core/Metadata/EdmLibraryExtensions.cs

```diff
@@ -263,6 +277,19 @@ internal static IEnumerable<IEdmOperation> FilterBoundOperationsWithSameTypeHier
         }
 
         /// <summary>
+        /// Filters the operations by parameter names.
+        /// </summary>
+        /// <param name="functions">The operations.</param>
+        /// <param name="parameters">The list of non-binding parameter names to match.</param>
+        /// <param name="caseInsensitive">Whether case insensitive.</param>
+        /// <returns>The best matching operations based on parameters.</returns>
+        internal static IEnumerable<IEdmOperation> FindBestOverloadBasedOnParameters(this IEnumerable<IEdmOperation> functions, IEnumerable<string> parameters, bool caseInsensitive = false)

```

Comment:
This is used to find the single best match operation overload based on multiple candidate overloads. The best match is defined as the one that has the exact same parameters. We need separate methods for Operations and OperationImports because these were written as extension methods that apply to and return those different collections types as part of a pipeline.

---
In reply to: [123296221](https://github.com/OData/odata.net/pull/828#discussion_r123296221) [](ancestors = 123296221)